homeassistant/components/daikin/switch.py

```diff
@@ -119,3 +124,41 @@ async def async_turn_on(self, **kwargs: Any) -> None:
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the zone off."""
         await self._api.device.set_streamer("off")
+
+
+class DaikinToggleSwitch(SwitchEntity):
+    """Switch state."""
+
+    _attr_icon = TOGGLE_ICON
+    _attr_has_entity_name = True
+
+    def __init__(self, daikin_api: DaikinApi) -> None:
+        """Initialize switch."""
+        self._api = daikin_api
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._api.device.mac}-toggle"
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the sensor."""
+        return ATTR_STATE_OFF not in self._api.device.represent(DAIKIN_ATTR_MODE)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        return self._api.device_info
+
+    async def async_update(self) -> None:
+        """Retrieve latest state."""
+        await self._api.async_update()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the zone on."""
+        await self._api.device.set({})
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the zone off."""
+        await self._api.device.set({DAIKIN_ATTR_MODE: ATTR_STATE_OFF})
```

Comment:
We should not pass our internal constants to external devices. If our constants would change, this would break :)

```suggestion
        await self._api.device.set({DAIKIN_ATTR_MODE: "off"})
```